src/IssueViz.Security/OpenInIDE/ListenerFactory.cs

```diff
@@ -73,5 +73,20 @@ HttpListener IListenerFactory.Create(int startPort, int endPort)
             }
             return listener;
         }
+
+        private static IDisposable CreateListener(int port)
+        {
+            var appProperties = new AppProperties(new Dictionary<string, object>())
+            {
+                Addresses = AddressCollection.Create()
+            };
+
+            var address = Address.Create();
+            address.Port = port.ToString();
+            appProperties.Addresses.Add(address);
+
+            var openInIDEListener = new OpenInIDEHttpListener();
+            return OwinServerFactory.Create(openInIDEListener.ProcessRequest, appProperties.Dictionary);
```

Comment:
At present we don't need an OWIN pipeline of components to handle auth, routing etc (and we probably never will). So here I'm directly invoking the specific server factory and passing it our handler class that will handle the requests as they arrive.